LibreHardwareMonitorLib/Hardware/Gpu/AmdGpu.cs

```diff
@@ -146,6 +157,44 @@ public AmdGpu(AtiAdlxx.ADLAdapterInfo adapterInfo, ISettings settings)
             ActivateSensor(_fullscreenFps);
         }
 
+        if(AtiAdlxx.ADL_Method_Exists(nameof(AtiAdlxx.ADL2_Adapter_PMLog_Support_Get))
+            && AtiAdlxx.ADL_Method_Exists(nameof(AtiAdlxx.ADL2_Device_PMLog_Device_Create))
+            && AtiAdlxx.ADL_Method_Exists(nameof(AtiAdlxx.ADL2_Adapter_PMLog_Start)))
+        {
+            AtiAdlxx.ADLPMLogSupportInfo _adlPMLogSupportInfo = new();
+            AtiAdlxx.ADLPMLogStartInput _adlPMLogStartInput = new();
+            _adlPMLogStartOutput = new();
+            _adlPMLogStartInput.usSensors = new ushort[AtiAdlxx.ADL_PMLOG_MAX_SENSORS];
+
+            AtiAdlxx.ADL2_Adapter_PMLog_Support_Get(_context, _adapterInfo.AdapterIndex, ref _adlPMLogSupportInfo);
+            if(_device == 0)
+            {
+                if(AtiAdlxx.ADLStatus.ADL_OK == AtiAdlxx.ADL2_Device_PMLog_Device_Create(_context, _adapterInfo.AdapterIndex, ref _device))
+                {
+                    int i = 0;
+                    while (_adlPMLogSupportInfo.usSensors[i] != (ushort) AtiAdlxx.ADLPMLogSensors.ADL_SENSOR_MAXTYPES)
+                    {
+                        _adlPMLogStartInput.usSensors[i] = _adlPMLogSupportInfo.usSensors[i];
+                        i++;
+                    }
+
+                    _adlPMLogStartInput.usSensors[i] = (ushort) AtiAdlxx.ADLPMLogSensors.ADL_SENSOR_MAXTYPES;
+                    _adlPMLogStartInput.ulSampleRate = pmLogSampleRate;
+
+                    if(AtiAdlxx.ADL2_Adapter_PMLog_Start(_context, adapterInfo.AdapterIndex,
+                                                         ref _adlPMLogStartInput, ref _adlPMLogStartOutput, _device) == AtiAdlxx.ADLStatus.ADL_OK)
+                                                         {
+                        _pmLogStarted = true;
+                    }
+
+                    if(_pmLogStarted && _adapterInfo.AdapterName.Contains("AMD Radeon RX 7"))
```

Comment:
I prefer to remove the adapter name check, and leave just the method name validation. The method name check in theory would be sufficient considering this pmlog support works with all cards. I think we will have a problem if the methods are present and are incompatible in some way with some generation.

The commit when this pmlog support was introduced on display-library repo dates from 2018 (ADL 11.0).

Reply:
I generally would like to avoid hardcoded names as it typically needs to be removed for new hardware generations. Best to just prefer this if valid data is returned, if not, fall back to previous method, etc (basically what's done now for cards without the PM stuff).

Reply:
I removed the name check, kept the _pmLogStarted bool  check only.